android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthManager.java

```diff
@@ -0,0 +1,630 @@
+    }
+
+    /**
+     * On success, clears the participant's email, password and session from storage
+     *
+     * @return notifies of completion or error
+     */
+    @NonNull
+    public Completable signOut() {
+        logger.debug("signOut called");
+
+        final String email;
+
+        SignIn signIn = accountDAO.getSignIn();
+        if (signIn != null) {
+            email = signIn.getEmail();
+        } else {
+            email = null;
+            logger.debug("Did not find saved SignIn credentials prior to calling sign out API");
+        }
+
+        return RxUtils.toBodySingle(authenticationApi.signOut())
+                .doOnSuccess(message -> {
+                    accountDAO.setSignIn(null);
+                    accountDAO.setUserSessionInfo(null);
+                    accountDAO.setStudyParticipant(null);
+
+                    for (AuthenticationEventListener listener : listeners) {
+                        listener.onSignedOut(email);
+                    }
+                }).toCompletable();
+    }
+
+    /**
+     * An email will be sent if an account is registered with the configured study.
+     *
+     * @param email participant's email address
+     * @return notifies of completion or error
+     */
+    @NonNull
+    public Completable requestPasswordReset(@NonNull String email) {
+        checkNotNull(email);
+
+        logger.debug("requestPasswordReset called with email: " + email);
+
+        return RxUtils.toBodySingle(authenticationApi.requestResetPassword(new Email().study
+                (config.getStudyId()).email(email))).toCompletable();
+    }
+
+    /**
+     * Get access to bridge API for currently authenticated client.
+     *
+     * @return API returns access to bridge that always uses the credentials currently held by
+     * AuthManager
+     */
+    @NonNull
+    public ForConsentedUsersApi getApi() {
+        logger.debug("getApi called");
+
+        return proxiedForConsentedUsersApi;
+    }
+
+    /**
+     * @return api access for currently logged in participant
+     */
+    @NonNull
+    ForConsentedUsersApi getRawApi() {
+        SignIn signIn = accountDAO.getSignIn();
+        if (signIn == null) {
+            return apiClientProvider.getClient(ForConsentedUsersApi.class);
+        }
+        return apiClientProvider.getClient(ForConsentedUsersApi.class, accountDAO.getSignIn());
+    }
+
+    /**
+     * @return the email currently associated with logged in participant
+     */
+    @Nullable
+    public String getEmail() {
+        SignIn signIn = accountDAO.getSignIn();
+        return signIn == null ? null : signIn.getEmail();
+    }
+
+    /**
+     * Get latest session from local store. Does not make a service call.
+     *
+     * @return Bridge session
+     */
+    @Nullable
+    public UserSessionInfo getUserSessionInfo() {
+        logger.debug("getUserSessionInfo called");
+
+        // TODO: a way to distinguish if session is null because we haven't signed on, or if it
```

Comment:
Is this still a valid TODO? We do need a way to check if the user has sign up but not signed in, that way the onboarding process will kick you back to email verification view if anything happened like the app crashed or they exited on the email verification screen